tools/chpl-language-server/src/chpl-language-server.py

```diff
@@ -1432,39 +1596,42 @@ def add_to_edits(nr: NodeAndRange):
 
     @server.feature(TEXT_DOCUMENT_INLAY_HINT)
     async def inlay_hint(ls: ChapelLanguageServer, params: InlayHintParams):
+        text_doc = ls.workspace.get_text_document(params.text_document.uri)
+        fi, _ = ls.get_file_info(text_doc.uri)
+        inlays: List[InlayHint] = []
+
         # The get_decl_inlays and get_call_inlays methods also check
         # and return early if the resolver is not being used, but for
         # the time being all hints are resolver-based, so we may
         # as well save ourselves the work of finding declarations and
         # calls to feed to those methods.
-        if not ls.use_resolver:
-            return None
-
-        text_doc = ls.workspace.get_text_document(params.text_document.uri)
-
-        fi, _ = ls.get_file_info(text_doc.uri)
-
-        decls = fi.def_segments.range(params.range)
-        calls = list(
-            call
-            for call, _ in chapel.each_matching(
-                fi.get_asts(), chapel.core.FnCall
+        if ls.use_resolver:
+            decls = fi.def_segments.range(params.range)
+            calls = list(
+                call
+                for call, _ in chapel.each_matching(
+                    fi.get_asts(), chapel.core.FnCall
+                )
             )
-        )
 
-        inlays: List[InlayHint] = []
-        with fi.context.context.track_errors() as _:
-            for decl in decls:
-                instantiation = fi.get_inst_segment_at_position(decl.rng.start)
-                inlays.extend(
-                    ls.get_decl_inlays(decl, fi.siblings, instantiation)
-                )
+            with fi.context.context.track_errors() as _:
+                for decl in decls:
+                    instantiation = fi.get_inst_segment_at_position(
+                        decl.rng.start
+                    )
+                    inlays.extend(ls.get_decl_inlays(decl, instantiation))
 
-            for call in calls:
-                instantiation = fi.get_inst_segment_at_position(
-                    location_to_range(call.location()).start
-                )
-                inlays.extend(ls.get_call_inlays(call, instantiation))
+                for call in calls:
+                    instantiation = fi.get_inst_segment_at_position(
+                        location_to_range(call.location()).start
+                    )
+                    inlays.extend(ls.get_call_inlays(call, instantiation))
+
+        file_lines = fi.file_lines()
+        ast = fi.get_asts()
+        block_inlays = ls.get_end_markers(ast, file_lines)
+        if len(block_inlays) > 0:
+            inlays.extend(block_inlays)
```

Comment:
To avoid indenting literally everything other than this code in the function behind a `use_resolver` conditional, consider moving this (non-resolver) stuff first, then returning `inlays` if `resolver` is not enabled, and proceeding through to modify inlays otherwise.